### PR TITLE
Improve I18n implementation

### PR DESCRIPTION
**Pull request by sisou**

### CODE REVIEW

File: src/components/LanguagePicker.js [hunk 1]
```diff
@@ -1,28 +1,37 @@
...
-            const label = i18n.translateWord('_language', language);
-            const selected = language == i18n.currentLanguage() ? ' selected' : '';
-            options.push(`<option value="${ language }"${ selected }>${ label }</option>`);
+        for (const language of this._i18n.availableLanguages()) {
+            const label = this._i18n.translatePhrase('_language', language);
+            const selectedAttr = language === this._i18n.language ? ' selected' : '';
+            options.push(`<option value="${ language }"${ selectedAttr }>${ label }</option>`);
         }
 
         element.innerHTML = options.join();
```
Comment (Bettelstab) [inline src/components/LanguagePicker.js]: replace innerHTML, because it's no constant string

File: src/lib/I18n.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+        /** @type {string} */
+        this._language;
+
+        this.setLanguage(language || navigator.language);
+    }
+
+    /**
+     * @param {HTMLElement} [dom] - The DOM element to be translated, or body by default
+     * @param {String} [lang] - ISO code of language to translate to, or the currently selected language by default
```
Comment (Bettelstab) [inline src/lib/I18n.js]: String -> string

**Comment by sisou** [comment]
@svub please add functionality to replace an input's `placeholder` attribute, I suggest via the `data-i18n-placeholder` marker.

**Comment by svub** [comment]
Ready to be merged!

File: src/lib/I18n.js [hunk 3]
```diff
@@ -64,17 +78,17 @@ class I18n {
...
         const languagePrefix = `${language}-`;
         for (const supportedLanguage of this.availableLanguages()) {
             if (supportedLanguage.startsWith(languagePrefix)) return supportedLanguage;
         }
-        // language is not supported. Return a variant of 'en' if available, otherwise an arbitrary supported language
-        if (language !== 'en') {
-            return this.getClosestSupportedLanguage('en');
+        // language is not supported. Return a variant of this._fallbackLanguage if available, otherwise an arbitrary supported language
+        if (language !== this._fallbackLanguage) {
+            return this.getClosestSupportedLanguage(this.fallbackLanguage);
```
Comment (danimoh) [inline src/lib/I18n.js]: will end up as endless recursion if fallback language has a hyphen, e.g. 'en-us' or if _fallbackLanguage is undefined

File: src/lib/I18n.js [hunk 4]
```diff
@@ -1,60 +1,74 @@
 class I18n {
+
     /**
-     * @param {object} dictionary
-     * @param {string} [language] - Language to translate to
+     * @param {object} dictionary - dictionary of all languages and phrases
+     * @param {string} fallbackLanguage - Language to be used if no translation for the current language can be found.
```
Comment (danimoh) [inline src/lib/I18n.js]: declare fallbackLanguage as optional and set to 'en' if not provided

File: src/lib/I18n.js [hunk 5]
```diff
@@ -83,19 +97,23 @@ class I18n {
...
     }
 
-    get dictionary() {
+    static get dictionary() {
         return this._dict;
     }
-}
+
+    static get fallbackLanguage() {
+        return this._fallbackLanguage || 'en';
```
Comment (danimoh) [inline src/lib/I18n.js]: set _fallbackLanguage to 'en' in constructor if fallbackLanguage was not provided instead of the " || 'en'" here

**Pull request merged**